driver-core/src/main/java/com/datastax/driver/core/Message.java

```diff
@@ -318,10 +325,12 @@ static class ProtocolEncoder extends MessageToMessageEncoder<Request> {
 
     @Override
     protected void encode(ChannelHandlerContext ctx, Request request, List<Object> out) {
+      ProtocolFeatureStore featureStore = ProtocolFeatureStore.loadFromChannel(ctx.channel());
```

Comment:
Update `ProtocolEncoder` to store `ProtocolFeatureStore`, `messageEncoderFor` to construct `ProtocolEncoder` on flight instead of silly static picker they currently have.
And update `switchToModernFraming` to rebuild pipeline with new encoder.